File: src/main/java/com/viaversion/aas/codec/packet/status/StatusPing.java
package com.viaversion.aas.codec.packet.status;

import com.viaversion.aas.codec.packet.Packet;
import io.netty.buffer.ByteBuf;
import org.jetbrains.annotations.NotNull;

public class StatusPing implements Packet {
	private long number;

	public long getNumber() {
		return number;
	}

	public void setNumber(long number) {
		this.number = number;
	}

	@Override
	public void decode(@NotNull ByteBuf byteBuf, int protocolVersion) throws Exception {
		number = byteBuf.readLong();
	}

	@Override
	public void encode(@NotNull ByteBuf byteBuf, int protocolVersion) throws Exception {
		byteBuf.writeLong(number);
	}
}

File: src/main/kotlin/com/viaversion/aas/packet/status/StatusPing.kt
package com.viaversion.aas.packet.status

import com.viaversion.aas.packet.Packet
import io.netty.buffer.ByteBuf
import kotlin.properties.Delegates

class StatusPing : Packet {
    var number by Delegates.notNull<Long>()
    override fun decode(byteBuf: ByteBuf, protocolVersion: Int) {
        number = byteBuf.readLong()
    }

    override fun encode(byteBuf: ByteBuf, protocolVersion: Int) {
        byteBuf.writeLong(number)
    }
}